shared/teams/team/invites/invite-item.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+import * as Container from '../../../util/container'
+import * as Types from '../../../constants/types/teams'
+import * as TeamsGen from '../../../actions/teams-gen'
+
+export type Mode = 'header' | 'settings' | 'history' | 'generate'
+
+export const InviteItem = ({
+  inviteLink,
+  teamID,
+  mode,
+}: {
+  inviteLink: Types.InviteLink
+  teamID: Types.TeamID
+  mode: Mode
+}) => {
+  const dispatch = Container.useDispatch()
+  const yourUsername = Container.useSelector(s => s.config.username)
+  const [waitingForExpire, setWaitingForExpire] = React.useState(false)
+  const onExpire = () => {
+    dispatch(TeamsGen.createRemovePendingInvite({inviteID: inviteLink.id, teamID}))
+
+    // wait until reload happens due to MetadataUpdate notification; otherwise it flashes
+    // active in between the rpc finish and the team reload
+    setWaitingForExpire(true)
```

Comment:
This is a new change I made.

Reply:
The rest of this function is pretty much the same except for some switching on the mode of where it's used, and swapping out expiration formatting with validityDescription